src/arbitrary.rs

```diff
@@ -1039,8 +1056,8 @@ impl Arbitrary for RangeFull {
 
 impl Arbitrary for Duration {
     fn arbitrary<G: Gen>(gen: &mut G) -> Self {
-        let seconds = u64::arbitrary(gen);
-        let nanoseconds = u32::arbitrary(gen) % 1_000_000;
+        let seconds = gen.gen_range(0, gen.size() as u64);
```

Comment:
Out of curiosity, why did you change this to use `gen.size()`? It was implemented this way before, but it seems like we should probably open up the range of durations generated too?

Reply:
I think this was an oversight and based on previous discussion, I think the criterion for whether a type respects `gen.size()` should be whether there are practical (memory, speed or some other resource) limits which make ignoring `gen.size()` impractical. Based on this criterion, `Duration` should not respect `gen.size()`. If you agree with this criterion, I could add it to the documentation for `Arbitrary`.